src/main/java/org/project/SaveState.java

```diff
@@ -1,34 +1,78 @@
 package org.project;
 
+import org.json.simple.JSONObject;
+
 /**
  * Defines a save object which stores the current game state in a file
  *
- * @author Nathan Bartyuk
- * @version 2023-02-07
+ * @author Nathan Bartyuk, Greg Song
+ * @version 2023-03-31
  */
 public class SaveState {
-  public enum tile {BG, WA, CH}
-  public enum sprite {BU, BO, RU}
-
-  public tile[][] tileMap;
-  public sprite[][] spriteMap;
-  private int playerX;
-  private int playerY;
-  private int playerRubies;
+  JSONObject playerData;
+  JSONObject gamePanelData;
 
-  public int x() {
-    return playerX;
+  /**
+   * Returns JSON data representing state of Player.
+   * @return a JSONObject
+   */
+  public JSONObject getPlayerData() {
+    return this.playerData;
   }
 
-  public int y() {
-    return playerY;
+  /**
+   * Returns JSON data representing state of GamePanel.
+   * @return a JSONObject
+   */
+  public JSONObject getGamePanelData() {
+    return this.gamePanelData;
   }
 
-  public int rubies() {
-    return playerRubies;
+  /**
+   * Sets SaveState data.
+   * @param player a Player instance
+   * @param gp a GamePanel instance
+   */
+  public void setSaveState(Player player, GamePanel gp) {
+    setPlayerData(player);
+    setGamePanelData(gp);
   }
 
-  public void save() {}
+  /* Helper methods */
+  /**
+   * Helper method to set the player data as a JSONObject.
+   * @param player, current instance of Player
+   */
+  private void setPlayerData(Player player) {
+    if (player == null) {
+      throw new NullPointerException("Player object is null.");
+    }
+    JSONObject playerData = new JSONObject();
+    playerData.put("worldX", player.worldX);
+    playerData.put("worldY", player.worldY);
+    playerData.put("screenX", player.screenX);
```

Comment:
These don't need to be saved as they are fixed <3
Character always drawn at the center of the screen